### PR TITLE
Fedora dependencies

### PR DESCRIPTION
**Pull request by aodhan-domhnaill**

### CODE REVIEW

**Comment by aidan-plenert-macdonald** [comment]
Fedora has some version increases from what you are using and the build fails

```
[aidan@aidan-laptop build]$ sudo dnf install cmake openssl-devel xerces-c-devel xml-security-c-devel zlib-devel podofo-devel https://www.codesynthesis.com/download/xsd/4.0/linux-gnu/x86_64/xsd-4.0.0-1.x86_64.rpm vim-common
[sudo] password for aidan: 
Last metadata expiration check: 0:20:14 ago on Tue 02 Mar 2021 07:13:46 PM PST.
xsd-4.0.0-1.x86_64.rpm                                                                                                        2.2 MB/s | 4.9 MB     00:02    
Package xsd-4.0.0-1.x86_64 is already installed.
Package cmake-3.18.4-2.fc33.x86_64 is already installed.
Package openssl-devel-1:1.1.1i-3.fc33.x86_64 is already installed.
Package xerces-c-devel-3.2.3-2.fc33.x86_64 is already installed.
Package xml-security-c-devel-2.0.2-5.fc33.x86_64 is already installed.
Package zlib-devel-1.2.11-23.fc33.x86_64 is already installed.
Package podofo-devel-0.9.6-13.fc33.x86_64 is already installed.
Package vim-common-2:8.2.2559-1.fc33.x86_64 is already installed.
Dependencies resolved.
```


## Errors

```
[aidan@aidan-laptop build]$ make -j
[  4%] Built target minizip
[ 47%] Built target digidocpp_priv
[ 51%] Building CXX object src/CMakeFiles/digidocpp.dir/SignatureCAdES_B.cpp.o
[ 51%] Building CXX object src/CMakeFiles/digidocpp.dir/SignatureCAdES_T.cpp.o
[ 51%] Building CXX object src/CMakeFiles/digidocpp.dir/PDF.cpp.o
In file included from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:21:
/home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_p.h:33:13: error: field ‘method’ has incomplete type ‘RSA_METHOD’ {aka ‘rsa_meth_st’}
   33 |  RSA_METHOD method;
      |             ^~~~~~
In file included from /usr/include/openssl/x509.h:15,
                 from /usr/include/openssl/cms.h:16,
                 from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_p.h:22,
                 from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:21:
/usr/include/openssl/ossl_typ.h:113:16: note: forward declaration of ‘RSA_METHOD’ {aka ‘struct rsa_meth_st’}
  113 | typedef struct rsa_meth_st RSA_METHOD;
      |                ^~~~~~~~~~~
In file included from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_T.cpp:21:
/home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_p.h:33:13: error: field ‘method’ has incomplete type ‘RSA_METHOD’ {aka ‘rsa_meth_st’}
   33 |  RSA_METHOD method;
      |             ^~~~~~
In file included from /usr/include/openssl/x509.h:15,
                 from /usr/include/openssl/cms.h:16,
                 from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_p.h:22,
                 from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_T.cpp:21:
/usr/include/openssl/ossl_typ.h:113:16: note: forward declaration of ‘RSA_METHOD’ {aka ‘struct rsa_meth_st’}
  113 | typedef struct rsa_meth_st RSA_METHOD;
      |                ^~~~~~~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp: In lambda function:
/home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:56:9: error: ‘memcpy’ was not declared in this scope
   56 |         memcpy(sigret, signature.data(), signature.size());
      |         ^~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:31:1: note: ‘memcpy’ is defined in header ‘<cstring>’; did you forget to ‘#include <cstring>’?
   30 | #include <openssl/ts.h>
  +++ |+#include <cstring>
   31 | 
/home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp: In constructor ‘digidoc::SignatureCAdES_B::SignatureCAdES_B(digidoc::Signer*)’:
/home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:66:8: error: invalid use of incomplete type ‘RSA’ {aka ‘struct rsa_st’}
   66 |     rsa->flags |= RSA_FLAG_SIGN_VER;
      |        ^~
In file included from /usr/include/openssl/x509.h:15,
                 from /usr/include/openssl/cms.h:16,
                 from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_p.h:22,
                 from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:21:
/usr/include/openssl/ossl_typ.h:112:16: note: forward declaration of ‘RSA’ {aka ‘struct rsa_st’}
  112 | typedef struct rsa_st RSA;
      |                ^~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:66:19: error: ‘RSA_FLAG_SIGN_VER’ was not declared in this scope
   66 |     rsa->flags |= RSA_FLAG_SIGN_VER;
      |                   ^~~~~~~~~~~~~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:72:30: error: invalid use of incomplete type ‘ESS_CERT_ID’ {aka ‘struct ESS_cert_id’}
   72 |     ASN1_OCTET_STRING_set(cid->hash, cert->sha1_hash, sizeof(cert->sha1_hash));
      |                              ^~
In file included from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:30:
/usr/include/openssl/ts.h:59:16: note: forward declaration of ‘ESS_CERT_ID’ {aka ‘struct ESS_cert_id’}
   59 | typedef struct ESS_cert_id ESS_CERT_ID;
      |                ^~~~~~~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:72:42: error: invalid use of incomplete type ‘using X509 = struct x509_st’ {aka ‘struct x509_s’}
   72 |     ASN1_OCTET_STRING_set(cid->hash, cert->sha1_hash, sizeof(cert->sha1_hash));
      |                                          ^~
In file included from /usr/include/openssl/x509.h:15,
                 from /usr/include/openssl/cms.h:16,
                 from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_p.h:22,
                 from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:21:
/usr/include/openssl/ossl_typ.h:123:16: note: forward declaration of ‘using X509 = struct x509_st’ {aka ‘struct x509_st’}
  123 | typedef struct x509_st X509;
      |                ^~~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:72:66: error: invalid use of incomplete type ‘using X509 = struct x509_st’ {aka ‘struct x509_s’}
   72 |     ASN1_OCTET_STRING_set(cid->hash, cert->sha1_hash, sizeof(cert->sha1_hash));
      |                                                                  ^~
In file included from /usr/include/openssl/x509.h:15,
                 from /usr/include/openssl/cms.h:16,
                 from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_p.h:22,
                 from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:21:
/usr/include/openssl/ossl_typ.h:123:16: note: forward declaration of ‘using X509 = struct x509_st’ {aka ‘struct x509_st’}
  123 | typedef struct x509_st X509;
      |                ^~~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:76:38: error: invalid use of incomplete type ‘using X509 = struct x509_st’ {aka ‘struct x509_s’}
   76 |     name->d.dirn = X509_NAME_dup(cert->cert_info->issuer);
      |                                      ^~
In file included from /usr/include/openssl/x509.h:15,
                 from /usr/include/openssl/cms.h:16,
                 from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_p.h:22,
                 from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:21:
/usr/include/openssl/ossl_typ.h:123:16: note: forward declaration of ‘using X509 = struct x509_st’ {aka ‘struct x509_st’}
  123 | typedef struct x509_st X509;
      |                ^~~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:78:8: error: invalid use of incomplete type ‘ESS_CERT_ID’ {aka ‘struct ESS_cert_id’}
   78 |     cid->issuer_serial = ESS_ISSUER_SERIAL_new();
      |        ^~
In file included from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:30:
/usr/include/openssl/ts.h:59:16: note: forward declaration of ‘ESS_CERT_ID’ {aka ‘struct ESS_cert_id’}
   59 | typedef struct ESS_cert_id ESS_CERT_ID;
      |                ^~~~~~~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:79:29: error: invalid use of incomplete type ‘ESS_CERT_ID’ {aka ‘struct ESS_cert_id’}
   79 |     sk_GENERAL_NAME_push(cid->issuer_serial->issuer, name);
      |                             ^~
In file included from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:30:
/usr/include/openssl/ts.h:59:16: note: forward declaration of ‘ESS_CERT_ID’ {aka ‘struct ESS_cert_id’}
   59 | typedef struct ESS_cert_id ESS_CERT_ID;
      |                ^~~~~~~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:80:26: error: invalid use of incomplete type ‘ESS_CERT_ID’ {aka ‘struct ESS_cert_id’}
   80 |     ASN1_INTEGER_free(cid->issuer_serial->serial);
      |                          ^~
In file included from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:30:
/usr/include/openssl/ts.h:59:16: note: forward declaration of ‘ESS_CERT_ID’ {aka ‘struct ESS_cert_id’}
   59 | typedef struct ESS_cert_id ESS_CERT_ID;
      |                ^~~~~~~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:81:8: error: invalid use of incomplete type ‘ESS_CERT_ID’ {aka ‘struct ESS_cert_id’}
   81 |     cid->issuer_serial->serial = ASN1_INTEGER_dup(cert->cert_info->serialNumber);
      |        ^~
In file included from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:30:
/usr/include/openssl/ts.h:59:16: note: forward declaration of ‘ESS_CERT_ID’ {aka ‘struct ESS_cert_id’}
   59 | typedef struct ESS_cert_id ESS_CERT_ID;
      |                ^~~~~~~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:81:55: error: invalid use of incomplete type ‘using X509 = struct x509_st’ {aka ‘struct x509_s’}
   81 |     cid->issuer_serial->serial = ASN1_INTEGER_dup(cert->cert_info->serialNumber);
      |                                                       ^~
In file included from /usr/include/openssl/x509.h:15,
                 from /usr/include/openssl/cms.h:16,
                 from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_p.h:22,
                 from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:21:
/usr/include/openssl/ossl_typ.h:123:16: note: forward declaration of ‘using X509 = struct x509_st’ {aka ‘struct x509_st’}
  123 | typedef struct x509_st X509;
      |                ^~~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:84:8: error: invalid use of incomplete type ‘ESS_SIGNING_CERT’ {aka ‘struct ESS_signing_cert’}
   84 |     ess->cert_ids = sk_ESS_CERT_ID_new_null();
      |        ^~
In file included from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:30:
/usr/include/openssl/ts.h:60:16: note: forward declaration of ‘ESS_SIGNING_CERT’ {aka ‘struct ESS_signing_cert’}
   60 | typedef struct ESS_signing_cert ESS_SIGNING_CERT;
      |                ^~~~~~~~~~~~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:85:28: error: invalid use of incomplete type ‘ESS_SIGNING_CERT’ {aka ‘struct ESS_signing_cert’}
   85 |     sk_ESS_CERT_ID_push(ess->cert_ids, cid);
      |                            ^~
In file included from /home/aidan/foss/digidoc/libdigidocpp/src/SignatureCAdES_B.cpp:30:
/usr/include/openssl/ts.h:60:16: note: forward declaration of ‘ESS_SIGNING_CERT’ {aka ‘struct ESS_signing_cert’}
   60 | typedef struct ESS_signing_cert ESS_SIGNING_CERT;
      |                ^~~~~~~~~~~~~~~~
make[2]: *** [src/CMakeFiles/digidocpp.dir/build.make:501: src/CMakeFiles/digidocpp.dir/SignatureCAdES_B.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [src/CMakeFiles/digidocpp.dir/build.make:514: src/CMakeFiles/digidocpp.dir/SignatureCAdES_T.cpp.o] Error 1
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp:48:1: error: expected class-name before ‘{’ token
   48 | {
      | ^
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp: In constructor ‘digidoc::DSSPdfSigIncMemDocument::DSSPdfSigIncMemDocument(const char*)’:
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp:50:58: error: class ‘digidoc::DSSPdfSigIncMemDocument’ does not have any field named ‘PdfSigIncMemDocument’
   50 |     DSSPdfSigIncMemDocument(const char* pszInpFilename): PdfSigIncMemDocument(pszInpFilename) {}
      |                                                          ^~~~~~~~~~~~~~~~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp: In member function ‘PoDoFo::PdfAcroForm* digidoc::DSSPdfSigIncMemDocument::GetAcroForm()’:
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp:51:41: error: ‘GetExistedAcroForm’ was not declared in this scope
   51 |     PdfAcroForm* GetAcroForm() { return GetExistedAcroForm(nullptr); }
      |                                         ^~~~~~~~~~~~~~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp: At global scope:
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp:173:6: error: no declaration matches ‘void digidoc::PDF::addDataFile(std::istream*, const string&, const string&)’
  173 | void PDF::addDataFile(istream * /*is*/, const string & /*fileName*/, const string & /*mediaType*/)
      |      ^~~
In file included from /home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp:20:
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.h:38:10: note: candidates are: ‘virtual void digidoc::PDF::addDataFile(std::unique_ptr<std::basic_istream<char> >, const string&, const string&)’
   38 |     void addDataFile(std::unique_ptr<std::istream> is, const std::string &fileName, const std::string &mediaType) override;
      |          ^~~~~~~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp:168:6: note:                 ‘virtual void digidoc::PDF::addDataFile(const string&, const string&)’
  168 | void PDF::addDataFile(const string & /*path*/, const string & /*mediaType*/)
      |      ^~~
In file included from /home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp:20:
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.h:30:7: note: ‘class digidoc::PDF’ defined here
   30 | class PDF: public Container
      |       ^~~
In file included from /home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp:24:
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp: In static member function ‘static std::unique_ptr<digidoc::Container> digidoc::PDF::openInternal(const string&)’:
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp:279:21: error: ‘EXCEPTION’ was not declared in this scope; did you mean ‘EXCEPTION_ADD’?
  279 |         THROW_CAUSE(EXCEPTION(e.what()), "Failed to parse PDF.");
      |                     ^~~~~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/log.h:63:81: note: in definition of macro ‘THROW_CAUSE’
   63 | #define THROW_CAUSE(_cause, ...) throw Exception(EXCEPTION_PARAMS(__VA_ARGS__), _cause)
      |                                                                                 ^~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp: In member function ‘virtual digidoc::Signature* digidoc::PDF::sign(digidoc::Signer*)’:
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp:324:13: error: ‘class digidoc::DSSPdfSigIncMemDocument’ has no member named ‘GetSignatureField’
  324 |         doc.GetSignatureField()->SetSignatureDate(PdfDate());
      |             ^~~~~~~~~~~~~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp:325:13: error: ‘class digidoc::DSSPdfSigIncMemDocument’ has no member named ‘GetSignatureField’
  325 |         doc.GetSignatureField()->SetSignatureReason(roles);
      |             ^~~~~~~~~~~~~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp:326:13: error: ‘class digidoc::DSSPdfSigIncMemDocument’ has no member named ‘Initialize’
  326 |         doc.Initialize();
      |             ^~~~~~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp:354:13: error: ‘class digidoc::DSSPdfSigIncMemDocument’ has no member named ‘Write’
  354 |         doc.Write(&out);
      |             ^~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp:376:90: error: no match for ‘operator=’ (operand types are ‘std::unique_ptr<std::basic_istream<char> >’ and ‘std::shared_ptr<std::__cxx11::basic_stringstream<char> >’)
  376 |         dataFile->m_is = make_shared<stringstream>(string(buf.GetBuffer(), buf.GetSize()));
      |                                                                                          ^
In file included from /usr/include/c++/10/memory:83,
                 from /home/aidan/foss/digidoc/libdigidocpp/src/Container.h:24,
                 from /home/aidan/foss/digidoc/libdigidocpp/src/PDF.h:22,
                 from /home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp:20:
/usr/include/c++/10/bits/unique_ptr.h:371:19: note: candidate: ‘std::unique_ptr<_Tp, _Dp>& std::unique_ptr<_Tp, _Dp>::operator=(std::unique_ptr<_Tp, _Dp>&&) [with _Tp = std::basic_istream<char>; _Dp = std::default_delete<std::basic_istream<char> >]’
  371 |       unique_ptr& operator=(unique_ptr&&) = default;
      |                   ^~~~~~~~
/usr/include/c++/10/bits/unique_ptr.h:371:29: note:   no known conversion for argument 1 from ‘std::shared_ptr<std::__cxx11::basic_stringstream<char> >’ to ‘std::unique_ptr<std::basic_istream<char> >&&’
  371 |       unique_ptr& operator=(unique_ptr&&) = default;
      |                             ^~~~~~~~~~~~
/usr/include/c++/10/bits/unique_ptr.h:386:2: note: candidate: ‘template<class _Up, class _Ep> typename std::enable_if<std::__and_<std::__and_<std::is_convertible<typename std::unique_ptr<_Up, _Ep>::pointer, typename std::__uniq_ptr_impl<_Tp, _Dp>::pointer>, std::__not_<std::is_array<_Up> > >, std::is_assignable<_T2&, _U2&&> >::value, std::unique_ptr<_Tp, _Dp>&>::type std::unique_ptr<_Tp, _Dp>::operator=(std::unique_ptr<_Up, _Ep>&&) [with _Up = _Up; _Ep = _Ep; _Tp = std::basic_istream<char>; _Dp = std::default_delete<std::basic_istream<char> >]’
  386 |  operator=(unique_ptr<_Up, _Ep>&& __u) noexcept
      |  ^~~~~~~~
/usr/include/c++/10/bits/unique_ptr.h:386:2: note:   template argument deduction/substitution failed:
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp:376:90: note:   ‘std::shared_ptr<std::__cxx11::basic_stringstream<char> >’ is not derived from ‘std::unique_ptr<_Tp, _Dp>’
  376 |         dataFile->m_is = make_shared<stringstream>(string(buf.GetBuffer(), buf.GetSize()));
      |                                                                                          ^
In file included from /usr/include/c++/10/memory:83,
                 from /home/aidan/foss/digidoc/libdigidocpp/src/Container.h:24,
                 from /home/aidan/foss/digidoc/libdigidocpp/src/PDF.h:22,
                 from /home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp:20:
/usr/include/c++/10/bits/unique_ptr.h:395:7: note: candidate: ‘std::unique_ptr<_Tp, _Dp>& std::unique_ptr<_Tp, _Dp>::operator=(std::nullptr_t) [with _Tp = std::basic_istream<char>; _Dp = std::default_delete<std::basic_istream<char> >; std::nullptr_t = std::nullptr_t]’
  395 |       operator=(nullptr_t) noexcept
      |       ^~~~~~~~
/usr/include/c++/10/bits/unique_ptr.h:395:17: note:   no known conversion for argument 1 from ‘std::shared_ptr<std::__cxx11::basic_stringstream<char> >’ to ‘std::nullptr_t’
  395 |       operator=(nullptr_t) noexcept
      |                 ^~~~~~~~~
In file included from /home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp:24:
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp:384:21: error: ‘EXCEPTION’ was not declared in this scope; did you mean ‘EXCEPTION_ADD’?
  384 |         THROW_CAUSE(EXCEPTION(e.what()), "Failed to sign PDF.");
      |                     ^~~~~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/log.h:63:81: note: in definition of macro ‘THROW_CAUSE’
   63 | #define THROW_CAUSE(_cause, ...) throw Exception(EXCEPTION_PARAMS(__VA_ARGS__), _cause)
      |                                                                                 ^~~~~~
/home/aidan/foss/digidoc/libdigidocpp/src/PDF.cpp:386:1: warning: control reaches end of non-void function [-Wreturn-type]
  386 | }
      | ^
make[2]: *** [src/CMakeFiles/digidocpp.dir/build.make:488: src/CMakeFiles/digidocpp.dir/PDF.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:190: src/CMakeFiles/digidocpp.dir/all] Error 2
make: *** [Makefile:160: all] Error 2
```

File: README.md [hunk 1]
```diff
@@ -11,11 +11,14 @@
...
 
-### Ubuntu
+### Ubuntu, Fedora
 
 1. Install dependencies
 
+        # Ubuntu
         sudo apt-get install cmake libxml-security-c-dev xsdcxx libssl-dev zlib1g-dev
+        # Fedora
+        sudo dnf install cmake openssl-devel xerces-c-devel xml-security-c-devel zlib-devel podofo-devel vim-common https://www.codesynthesis.com/download/xsd/4.0/linux-gnu/x86_64/xsd-4.0.0-1.x86_64.rpm
```
Comment (metsma) [inline README.md:21]: Please, remove podofo. They broke API and this code is unmaintained in digidocpp.

**Comment by aidan-plenert-macdonald** [comment]
It built for me on Fedora 33. Running the Github Action would be a useful check. If that works, I can work on the Fedora RPM.

**Comment by glensc** [comment]
Why do you need vim-common for non-interactive builds like CI?

**Comment by aidan-plenert-macdonald** [comment]
> Why do you need vim-common for non-interactive builds like CI?

@glensc I just copied the Ubuntu deps. I can remove it.

**Comment by aidan-plenert-macdonald** [comment]
The RPM doesn't install yet. I had the same problems when I used `alien` to install the currently supported `deb` packages on my computer.

```
[aidan@localhost build]$ sudo dnf localinstall -y ./libdigidocpp-3.14.7.0-1.x86_64.rpm 
  ...
Running transaction check
Transaction check succeeded.
Running transaction test
Error: Transaction test error:
  file /usr/local from install of libdigidocpp-3.14.7.0-1.x86_64 conflicts with file from package filesystem-3.14-3.fc33.x86_64
  file /usr/local/bin from install of libdigidocpp-3.14.7.0-1.x86_64 conflicts with file from package filesystem-3.14-3.fc33.x86_64
  file /usr/local/etc from install of libdigidocpp-3.14.7.0-1.x86_64 conflicts with file from package filesystem-3.14-3.fc33.x86_64
  file /usr/local/include from install of libdigidocpp-3.14.7.0-1.x86_64 conflicts with file from package filesystem-3.14-3.fc33.x86_64
  file /usr/local/lib64 from install of libdigidocpp-3.14.7.0-1.x86_64 conflicts with file from package filesystem-3.14-3.fc33.x86_64
  file /usr/local/share from install of libdigidocpp-3.14.7.0-1.x86_64 conflicts with file from package filesystem-3.14-3.fc33.x86_64
  file /usr/local/share/man from install of libdigidocpp-3.14.7.0-1.x86_64 conflicts with file from package filesystem-3.14-3.fc33.x86_64
  file /usr/local/share/man/man1 from install of libdigidocpp-3.14.7.0-1.x86_64 conflicts with file from package filesystem-3.14-3.fc33.x86_64
```

Should the deb also use CPack?

**Comment by aidan-plenert-macdonald** [comment]
@metsma The RPM generated now installs fine in Fedora 33. Only potential issues is that it includes the `/usr/include` files that would normally go into the `*-devel` package. I couldn't figure out how to separate them.

File: CMakeLists.txt [hunk 2]
```diff
@@ -30,6 +30,16 @@ set( BUILD_TOOLS YES CACHE BOOL "Build digidoc-tool" )
...
 set( SIGNCERT "" CACHE STRING "Common name of certificate to used sign binaries, empty skip signing" )
 set( CROSSSIGNCERT "" CACHE STRING "Common name of certificate to used cross sign binaries, empty skip signing" )
+set(CPACK_PACKAGE_VERSION ${VERSION})
+set(CPACK_GENERATOR "RPM" "DEB")
+set(CPACK_PACKAGE_NAME "libdigidocpp")
+set(CPACK_PACKAGE_RELEASE 1)
+set(CPACK_PACKAGE_CONTACT "Andrus Randveer")
+set(CPACK_PACKAGE_VENDOR "RIA")
+set(CPACK_PACKAGING_INSTALL_PREFIX "/usr/")
+set(CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "/usr/local" "/usr/local/etc" "/usr/share/man" "/usr/share/man/man1" "/usr/lib64/pkgconfig")
```
Comment (metsma) [inline CMakeLists.txt:40]: I don't thing package should install anything into /usr/local folder

**Comment by metsma** [comment]
Can you fix conflict

**Pull request closed without merging**